src/axolotl/state.rs

```diff
@@ -130,16 +129,9 @@ impl <T:Axolotl> ReceiveChain<T> {
         return Some(self.message_keys.pop().unwrap().1);
     }
 }
-impl <T:Axolotl> AxolotlState<T> {
 
-    pub fn encrypt(&mut self, plaintext : &T::PlainText) -> (AxolotlMessage<T>, T::Mac) {
-        let mut self_clone = Clone::clone(self);
-        let result = self_clone.try_encrypt(plaintext);
-        *self = self_clone;
-        result
-    }
-
-    fn try_encrypt(&mut self, plaintext : &T::PlainText) -> (AxolotlMessage<T>, T::Mac) {
```

Comment:
encrypt can't fail here, so you're correct that doing the copy isn't needed.

it's not obvious enough that a possibly panic!ing function shouldn't be called when the AxolotlState is in an inconsistent state, though. https://github.com/mjewkes/Axolotl-Rust/issues/14